drivers/soundwire/bus.c

```diff
@@ -1257,13 +1257,21 @@ static int sdw_slave_set_frequency(struct sdw_slave *slave)
 static int sdw_initialize_slave(struct sdw_slave *slave)
 {
 	struct sdw_slave_prop *prop = &slave->prop;
+	int status;
 	int ret;
 	u8 val;
 
 	ret = sdw_slave_set_frequency(slave);
 	if (ret < 0)
 		return ret;
 
+	/* Clear bus clash interrupt before enabling interrupt mask */
+	status = sdw_read_no_pm(slave, SDW_SCP_INT1);
+	if (status & SDW_SCP_INT1_BUS_CLASH) {
+		dev_warn(&slave->dev, "Bus clash detected before enabling\n");
+		sdw_write_no_pm(slave, SDW_SCP_INT1, SDW_SCP_INT1_BUS_CLASH);
```

Comment:
sdw_update_no_pm ?

Reply:
> sdw_update_no_pm ?

I got it . update can't work for it will compare clash bit. But how about other bits in SCP_INT1?

Reply:
> > sdw_update_no_pm ?
> 
> I got it . update can't work for it will compare clash bit. But how about other bits in SCP_INT1?

We only want to clear the bus clash bit, so set 0 to all other bits.